src/index.ts

```diff
@@ -0,0 +1,149 @@
+const DEFAULT_RESIZE_DEBOUNCE = 50;
+
+interface DeviceInfo<devices> {
+    size: number;
+    name: keyof devices;
+    index: number;
+}
+
+interface importDevices {
+    [key: string]: number;
+}
+
+interface ObserverOptions {
+    resizeDebounce?: number;
+    mobileFirst?: boolean;
+}
+
+export enum operators {
+    equal = '=',
+    more = '>',
+    moreOrEqual = '>=',
+    less = '<',
+    lessOrEqual = '<='
+}
+
+/**
+ * @param {importDevices} devices - Collection of devices and its breakpoints from which they start
+ * For example: {
+ *   'mobile': 0,
+ *   'tablet': 768,
+ *   'desktop': 1024,
+ * }
+ * @param {ObserverOptions} options
+ */
+
+export class DeviceObserver<Devices extends importDevices> {
+    private _devices: DeviceInfo<Devices>[] = [];
+    private _deviceChangeCallbacks: CallableFunction[] = [];
+    private _resizeCallbacks: CallableFunction[] = [];
+    private _debounceTimeoutId: number = 0;
+    private _currentDevice: DeviceInfo<Devices> = { size: 0, name: '', index: 0 };
+    private _options: ObserverOptions = { resizeDebounce: DEFAULT_RESIZE_DEBOUNCE, mobileFirst: true };
+
+    constructor(devices: Devices, options?: ObserverOptions) {
+        this._devices = this._sortDevices(devices);
+        if (options) this._options = Object.assign(this._options, options);
+        const device = this._getCurrentDevice;
+        if (device) this._currentDevice = device;
+        window.addEventListener('resize', this._onWindowResize);
```

Comment:
@Ofigelov as a performance improvement can we look into changing `resize` listener with `matchMedia.onchange`, see [this approach](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Testing_media_queries#Receiving_query_notifications) and [this](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/onchange)
this way we can remove debounce and do perform checks only against real media query change

Reply:
For the first one, `change` event of matchMedia doesn't available in IE11
For the second one, I think that having the ability to subscribe singlepointly on viewport resize can be helpfull in different situations, for example, to keep the variable `--viewport-height` with actual value.